crates/rattler_repodata_gateway/src/fetch/jlap/mod.rs

```diff
@@ -537,13 +536,13 @@ fn apply_jlap_patches(
         reporter.on_jlap_decode_start(index);
     }
 
+    tracing::info!("parsing cached repodata.json as JSON");
     // Read the contents of the current repodata to a string
-    let repo_data_contents = fs::read_to_string(repo_data_path).map_err(JLAPError::FileSystem)?;
+    let repo_data_file = fs::File::open(repo_data_path).map_err(JLAPError::FileSystem)?;
+    let repo_data_reader = std::io::BufReader::with_capacity(64 * 1024, repo_data_file);
 
-    // Parse the JSON so we can manipulate it
-    tracing::info!("parsing cached repodata.json as JSON");
-    let mut repo_data =
-        serde_json::from_str::<Value>(&repo_data_contents).map_err(JLAPError::JSONParse)?;
+    let mut repo_data: Value =
+        serde_json::from_reader(repo_data_reader).map_err(JLAPError::JSONParse)?;
```

Comment:
My fear was that loading the entire repodata into memory is about 80MB uncompressed which is quite a lot of memory in a memory constrained environment.

Good point about the performance. In fact, I encountered the same problem in the distant past and had just forgotten about it :pensive: 

Memory-mapping the file is likely the best compromise since this will allow us to use the fast read_string and also allows the kernel to freely unload pages under memory pressure.

Reply:
Does rattler implement any locking mechanism that would protect us from race conditions when using memory mapped files?

Reply:
It does yes. There is a file lock that should prevent multiple processes to access the same file.

Reply:
I've decided to revert memmap changes and buffered reader changes and swap to std::mem::drop, since it is the cheapest to implement for the time I have left to spend on this particular issue.